ls1intum/Artemis
# `Exam mode`: Make browser fingerprints configurable for administrators

### Checklist
#### General


#### Client

### Motivation and Context



Fixes #4913 

### Description


Adds a new flag to the ``/info`` endpoint to enable or disable browser fingerprinting; if disabled, the fingerprint service will clear the existing instance key from the local storage and not collect any fingerprints

### Steps for Testing


This is best tested locally as you have full control over your server config.
Run a few requests from the client and make sure that fingerprint headers are only submitted if it's enabled on the server side:

1. Three cases: In ``application.yml``, edit ``browser-fingerprints-enabled`` to ``true`` or ``false``, or remove the configuration option entirely.
2. Restart the server and reload the client
3. Make sure that the headers ``X-Artemis-Client-Instance-ID`` and ``X-Artemis-Client-Fingerprint`` are only submitted in subsequent rest calls if the configuration item is set to ``true`` or has not been defined (to not break current configurations); they should not be submitted if the config is set to ``false``

### Review Progress





#### Code Review
#### Manual Tests
